<a href="https://colab.research.google.com/github/sejalvyas13/Basic-CYK-Parser/blob/master/IWSLT_baseline.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# IWSLT Baseline

In [ ]:
!git clone https://github.com/abdulrafae/nmt_tutorial.git

Cloning into 'nmt_tutorial'...
remote: Enumerating objects: 88, done.
remote: Counting objects: 100% (88/88), done.
remote: Compressing objects: 100% (87/87), done.
remote: Total 88 (delta 46), reused 0 (delta 0), pack-reused 0
Unpacking objects: 100% (88/88), done.


In [ ]:
cd nmt_tutorial/

/content/nmt_tutorial


In [ ]:
!pip install fairseq


     |████████████████████████████████| 1.7MB 13.9MB/s 
     |████████████████████████████████| 133kB 51.7MB/s 
     |████████████████████████████████| 61kB 9.3MB/s 
     |████████████████████████████████| 112kB 49.9MB/s 
     |████████████████████████████████| 645kB 23.2MB/s 
  Created wheel for antlr4-python3-runtime: filename=antlr4_python3_runtime-4.8-cp37-none-any.whl size=141231 sha256=8de6ad00af3f56e663b33e7ee5fa91c330a278b84518a27f31cfa6ba0fc24e34
  Stored in directory: /root/.cache/pip/wheels/e3/e2/fa/b78480b448b8579ddf393bebd3f47ee23aa84c89b6a78285c8
Successfully built antlr4-python3-runtime
  Found existing installation: PyYAML 3.13
    Uninstalling PyYAML-3.13:
      Successfully uninstalled PyYAML-3.13


In [ ]:
!git clone https://github.com/moses-smt/mosesdecoder.git


Cloning into 'mosesdecoder'...
remote: Enumerating objects: 148070, done.
remote: Counting objects: 100% (498/498), done.
remote: Compressing objects: 100% (206/206), done.
remote: Total 148070 (delta 315), reused 433 (delta 289), pack-reused 147572
Receiving objects: 100% (148070/148070), 129.86 MiB | 23.08 MiB/s, done.
Resolving deltas: 100% (114341/114341), done.


In [ ]:
!git clone https://github.com/rsennrich/subword-nmt.git


Cloning into 'subword-nmt'...
remote: Enumerating objects: 580, done.
remote: Counting objects: 100% (4/4), done.
remote: Compressing objects: 100% (4/4), done.
remote: Total 580 (delta 0), reused 0 (delta 0), pack-reused 576
Receiving objects: 100% (580/580), 237.41 KiB | 8.19 MiB/s, done.
Resolving deltas: 100% (349/349), done.


In [ ]:
!git clone https://github.com/pytorch/fairseq

Cloning into 'fairseq'...
remote: Enumerating objects: 27747, done.
remote: Counting objects: 100% (168/168), done.
remote: Compressing objects: 100% (91/91), done.
remote: Total 27747 (delta 87), reused 120 (delta 77), pack-reused 27579
Receiving objects: 100% (27747/27747), 11.59 MiB | 15.43 MiB/s, done.
Resolving deltas: 100% (20906/20906), done.


In [ ]:
cd fairseq


/content/nmt_tutorial/fairseq


In [ ]:
!pip install --editable .

Obtaining file:///content/nmt_tutorial/fairseq
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
    Preparing wheel metadata ... done
  Found existing installation: fairseq 0.10.2
    Uninstalling fairseq-0.10.2:
      Successfully uninstalled fairseq-0.10.2
  Running setup.py develop for fairseq


In [ ]:
cd ..

/content/nmt_tutorial


In [ ]:
!bash prepare_data.sh


In [ ]:
!bash train_fconv.sh


2021-04-30 22:29:26 | INFO | fairseq_cli.preprocess | Namespace(align_suffix=None, alignfile=None, all_gather_list_size=16384, azureml_logging=False, bf16=False, bpe=None, cpu=False, criterion='cross_entropy', dataset_impl='mmap', destdir='data-bin/iwslt17_fr_en', empty_cache_freq=0, fp16=False, fp16_init_scale=128, fp16_no_flatten_grads=False, fp16_scale_tolerance=0.0, fp16_scale_window=None, joined_dictionary=False, log_file=None, log_format=None, log_interval=100, lr_scheduler='fixed', memory_efficient_bf16=False, memory_efficient_fp16=False, min_loss_scale=0.0001, model_parallel_size=1, no_progress_bar=False, nwordssrc=-1, nwordstgt=-1, only_source=False, optimizer=None, padding_factor=8, plasma_path='/tmp/plasma', profile=False, quantization_config_path=None, reset_logging=False, scoring='bleu', seed=1, source_lang='fr', srcdict=None, suppress_crashes=False, target_lang='en', task='translation', tensorboard_logdir=None, testpref=None, tgtdict=None, threshold_loss_scale=None, thres

In [ ]:
mkdir output

In [ ]:
!fairseq-interactive data-bin/iwslt17_fr_en/ \
    --path checkpoint/iwslt17_fr_en/checkpoint_best.pt \
    --buffer-size 2000 --batch-size 128 \
    --beam 5 --remove-bpe \
    --input iwslt17.tokenized.fr-en/test.fr | tee output/iwslt17.test.fr-en.en.sys

Streaming output truncated to the last 5000 lines.
W-455	0.002	seconds
H-455	-1.6511956453323364	now , it &apos;s a conflict .
D-455	-1.6511956453323364	now , it &apos;s a conflict .
P-455	-3.4501 -0.7746 -3.5054 -1.3304 -3.0006 -0.4779 -0.5258 -0.1447
S-456	<unk> le procrastinateur , ce conflit a tendance à se terminer de la même manière à chaque fois , le laissant passer beaucoup de temps dans cette zone orange , un endroit facile et amusant , totalement hors du <unk> de la <unk> .
W-456	0.010	seconds
H-456	-1.0792527198791504	in order to get to the procrastinator , this conflict tends to end up in the same way every time , letting it spend a lot of time in that orange area , an easy and fun place , totally out of the gun &apos;s department .
D-456	-1.0792527198791504	in order to get to the procrastinator , this conflict tends to end up in the same way every time , letting it spend a lot of time in that orange area , an easy and fun place , totally out of the gun &apos;s department .

In [ ]:
from google.colab import files
files.download('output/iwslt17.test.fr-en.en.sys')

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

In [ ]:
!grep ^H output/iwslt17.test.fr-en.en.sys | cut -f3- > output/hypo.tok.sys


In [ ]:
!cat output/hypo.tok.sys | mosesdecoder/scripts/recaser/detruecase.perl | mosesdecoder/scripts/tokenizer/detokenizer.perl -l en > output/hypo.sys


Detokenizer Version $Revision: 4134 $
Language: en


**On IWSLT17 test data**





In [ ]:
#cat output/hypo.sys | sacrebleu --test-set iwslt17 --language-pair fr-en --tokenize=13a
!sacrebleu --tokenize=13a iwslt17.tokenized.fr-en/test.en < output/hypo.sys

BLEU+case.mixed+numrefs.1+smooth.exp+tok.13a+version.1.5.1 = 21.0 51.8/28.2/17.1/10.8 (BP = 0.922 ratio = 0.925 hyp_len = 28097 ref_len = 30374)


In [ ]:
!cp output/hypo.sys IWSLT_baseline.txt

In [ ]:
files.download('IWSLT_baseline.txt')

<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

**Result**




BLEU+case.mixed+numrefs.1+smooth.exp+tok.13a+version.1.4.14 = 21.4 52.2/28.2/17.2/10.8 (BP = 0.936 ratio = 0.938 hyp_len = 28501 ref_len = 30374)


**On MTNT test data**

In [ ]:
!wget https://github.com/pmichel31415/mtnt/releases/download/v1.1/MTNT.1.1.tar.gz

--2020-12-15 22:23:44--  https://github.com/pmichel31415/mtnt/releases/download/v1.1/MTNT.1.1.tar.gz
Resolving github.com (github.com)... 192.30.255.112
Connecting to github.com (github.com)|192.30.255.112|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://github-production-release-asset-2e65be.s3.amazonaws.com/125262764/1b3bee80-1e45-11e9-90db-a9db0f519608?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=AKIAIWNJYAX4CSVEH53A%2F20201215%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20201215T222345Z&X-Amz-Expires=300&X-Amz-Signature=a4d91c0a49b04f067b9e838c71935f1db1b4c883070a68edb75b9b1e72486db9&X-Amz-SignedHeaders=host&actor_id=0&key_id=0&repo_id=125262764&response-content-disposition=attachment%3B%20filename%3DMTNT.1.1.tar.gz&response-content-type=application%2Foctet-stream [following]
--2020-12-15 22:23:45--  https://github-production-release-asset-2e65be.s3.amazonaws.com/125262764/1b3bee80-1e45-11e9-90db-a9db0f519608?X-Amz-Algorithm=AWS4-HMAC-SHA25

In [ ]:
!tar -xvzf MTNT.1.1.tar.gz

MTNT/
MTNT/README.md
MTNT/monolingual/
MTNT/monolingual/dev.en
MTNT/monolingual/dev.fr
MTNT/monolingual/dev.ja
MTNT/monolingual/dev.tok.en
MTNT/monolingual/dev.tok.fr
MTNT/monolingual/dev.tok.ja
MTNT/monolingual/train.en
MTNT/monolingual/train.fr
MTNT/monolingual/train.ja
MTNT/monolingual/train.tok.en
MTNT/monolingual/train.tok.fr
MTNT/monolingual/train.tok.ja
MTNT/split_tsv.sh
MTNT/test/
MTNT/test/test.en-fr.tsv
MTNT/test/test.en-ja.tsv
MTNT/test/test.fr-en.tsv
MTNT/test/test.ja-en.tsv
MTNT/train/
MTNT/train/train.en-fr.tsv
MTNT/train/train.en-ja.tsv
MTNT/train/train.fr-en.tsv
MTNT/train/train.ja-en.tsv
MTNT/valid/
MTNT/valid/valid.en-fr.tsv
MTNT/valid/valid.en-ja.tsv
MTNT/valid/valid.fr-en.tsv
MTNT/valid/valid.ja-en.tsv


In [ ]:
!bash MTNT/split_tsv.sh

In [ ]:
!fairseq-interactive data-bin/iwslt17_fr_en/ \
    --path checkpoint/iwslt17_fr_en/checkpoint_best.pt \
    --buffer-size 2000 --batch-size 128 \
    --beam 5 --remove-bpe \
    --input MTNT/test/test.fr-en.fr | tee output/mtnt.test.fr-en.en.sys

Streaming output truncated to the last 5000 lines.
W-22	0.018	seconds
H-22	-2.2868094444274902	now , in the day , they said , &quot; let &apos;s put it through a procedure of markets .
D-22	-2.2868094444274902	now , in the day , they said , &quot; let &apos;s put it through a procedure of markets .
P-22	-2.4564 -0.3787 -3.7236 -3.2768 -1.7115 -1.1730 -2.9431 -0.9149 -1.2839 -2.2942 -3.4218 -0.6710 -4.3482 -2.0733 -2.8251 -0.6385 -5.6549 -1.1971 -5.9210 -0.9615 -0.1548
S-23	<unk> coup <unk> quand ils font leur <unk> <unk> <unk> une société privée qui dépêche un type pour faire 60 bornes avec la bouffe et le café.
W-23	0.018	seconds
H-23	-2.3750874996185303	so let &apos;s call it a school , a private company that &apos;s rushing to a guy to do 60 grossing , with the food and the flirtation .
D-23	-2.3750874996185303	so let &apos;s call it a school , a private company that &apos;s rushing to a guy to do 60 grossing , with the food and the flirtation .
P-23	-5.1095 -4.3127 -0.8413 -2.7786 

In [ ]:
!grep ^H output/mtnt.test.fr-en.en.sys | cut -f3- > output/mtnt_hypo.tok.sys


In [ ]:
!cat output/mtnt_hypo.tok.sys | mosesdecoder/scripts/recaser/detruecase.perl | mosesdecoder/scripts/tokenizer/detokenizer.perl -l en > output/mtnt_hypo.sys


Detokenizer Version $Revision: 4134 $
Language: en


In [ ]:
!sacrebleu --tokenize=13a MTNT/test/test.fr-en.en < output/mtnt_hypo.sys

BLEU+case.mixed+numrefs.1+smooth.exp+tok.13a+version.1.4.14 = 7.5 30.7/10.0/4.5/2.2 (BP = 1.000 ratio = 1.106 hyp_len = 20773 ref_len = 18777)


BLEU+case.mixed+numrefs.1+smooth.exp+tok.13a+version.1.4.14 = 7.5 30.7/10.0/4.5/2.2 (BP = 1.000 ratio = 1.106 hyp_len = 20773 ref_len = 18777)

In [ ]:
!zip -r output.zip output/

  adding: output/ (stored 0%)
  adding: output/mtnt_hypo.sys (deflated 70%)
  adding: output/mtnt_hypo.tok.sys (deflated 73%)
  adding: output/iwslt17.test.fr-en.en.sys (deflated 69%)
  adding: output/hypo.tok.sys (deflated 69%)
  adding: output/mtnt.test.fr-en.en.sys (deflated 69%)
  adding: output/hypo.sys (deflated 66%)


In [ ]:
!zip -r data-bin.zip data-bin/

  adding: data-bin/ (stored 0%)
  adding: data-bin/iwslt17_fr_en/ (stored 0%)
  adding: data-bin/iwslt17_fr_en/train.fr-en.en.idx (deflated 77%)
  adding: data-bin/iwslt17_fr_en/train.fr-en.fr.bin (deflated 65%)
  adding: data-bin/iwslt17_fr_en/dict.fr.txt (deflated 66%)
  adding: data-bin/iwslt17_fr_en/train.fr-en.fr.idx (deflated 75%)
  adding: data-bin/iwslt17_fr_en/valid.fr-en.fr.idx (deflated 73%)
  adding: data-bin/iwslt17_fr_en/valid.fr-en.en.idx (deflated 75%)
  adding: data-bin/iwslt17_fr_en/valid.fr-en.en.bin (deflated 41%)
  adding: data-bin/iwslt17_fr_en/train.fr-en.en.bin (deflated 42%)
  adding: data-bin/iwslt17_fr_en/dict.en.txt (deflated 63%)
  adding: data-bin/iwslt17_fr_en/preprocess.log (deflated 56%)
  adding: data-bin/iwslt17_fr_en/valid.fr-en.fr.bin (deflated 65%)
  adding: data-bin/iwslt17_fr^M_en^M^M^M/ (stored 0%)


In [ ]:
!zip -r checkpoint.zip checkpoint/

  adding: checkpoint/ (stored 0%)
  adding: checkpoint/iwslt17_fr_en/ (stored 0%)
  adding: checkpoint/iwslt17_fr_en/checkpoint2.pt (deflated 7%)
  adding: checkpoint/iwslt17_fr_en/checkpoint10.pt (deflated 6%)
  adding: checkpoint/iwslt17_fr_en/checkpoint17.pt (deflated 6%)
  adding: checkpoint/iwslt17_fr_en/checkpoint25.pt (deflated 6%)
  adding: checkpoint/iwslt17_fr_en/checkpoint18.pt (deflated 6%)
  adding: checkpoint/iwslt17_fr_en/checkpoint28.pt (deflated 6%)
  adding: checkpoint/iwslt17_fr_en/checkpoint33.pt (deflated 6%)
  adding: checkpoint/iwslt17_fr_en/checkpoint9.pt (deflated 6%)
  adding: checkpoint/iwslt17_fr_en/checkpoint20.pt (deflated 6%)
  adding: checkpoint/iwslt17_fr_en/checkpoint14.pt (deflated 6%)
  adding: checkpoint/iwslt17_fr_en/checkpoint13.pt (deflated 6%)
  adding: checkpoint/iwslt17_fr_en/checkpoint27.pt (deflated 6%)
  adding: checkpoint/iwslt17_fr_en/checkpoint31.pt (deflated 6%)
  adding: checkpoint/iwslt17_fr_en/checkpoint26.pt (deflated 6%)
  adding: 